### PR TITLE
[Dashboard] ERC20 Claim form

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
The focus of this PR is to refactor the token claim functionality by replacing the Drawer with a Sheet component and updating the UI elements and logic.

### Detailed summary
- Replaced `Drawer` with `Sheet` component for token claim form
- Updated UI elements like buttons, icons, form controls
- Added form validation and error handling logic
- Improved transaction handling and tracking functionality

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4669?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4669** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4669?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4662** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4662?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4661** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4661?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4643** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4643?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kien-ngo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/contract-ui/tabs/tokens/components/claim-button.tsx [hunk 1]
```diff
@@ -1,37 +1,171 @@
...
+                account,
+              });
+
+              if (approveTx) {
+                try {
+                  await sendAndConfirmTransaction.mutateAsync(approveTx);
+                } catch {
+                  return toast.error("Error approving ERC20 token");
+                }
+              }
```
Comment (jnsdls) [inline apps/dashboard/src/contract-ui/tabs/tokens/components/claim-button.tsx:132]: you can do the same `toast.promise` here while u wait for this to be confirmed (that way user sees what's going on)
```
const promise = sendAndConfirmTransaction.mutateAsync(approveTx);
toast.promise(promise, {...})
await promise // to actually wait for the promise to finish at runtime
...
```

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/contract-ui/tabs/tokens/components/claim-button.tsx (hunk 1 above)
Comment (kien-ngo) [reply apps/dashboard/src/contract-ui/tabs/tokens/components/claim-button.tsx]: ty ser. I was thinking about how I would want to execute this. Updated the code. Let's see what you think about it ^

File: apps/dashboard/src/contract-ui/tabs/tokens/components/claim-button.tsx [hunk 2]
```diff
@@ -1,37 +1,188 @@
...
+                    },
+                  },
+                );
+                toast.promise(promise, {
+                  loading: "Approving ERC20 tokens for this claim",
+                  success: "Tokens approved successfully",
+                  error: "Failed to approve token",
+                });
+
+                await promise;
```
Comment (jnsdls) [inline apps/dashboard/src/contract-ui/tabs/tokens/components/claim-button.tsx:144]: this should throw if it fails meaning you shouldn't need the separate `approvedSuccessful` - can easily test that by executing this and rejecting the approval TX in ur wallet, which should trigger the throw

**Review by jnsdls — CHANGES_REQUESTED** [review]

**Review by jnsdls — APPROVED** [review]

**Pull request merged**